subsys/bluetooth/host/gatt.c

```diff
@@ -720,7 +720,7 @@ static int db_hash_setup(struct gen_hash_state *state, uint8_t *key)
 		LOG_ERR("Unable to import the key for AES CMAC %d", ret);
 		return -EIO;
 	}
-	state->operation = psa_mac_operation_init();
+	memset(&state->operation, 0, sizeof(state->operation));
```

Comment:
Out of curiosity, where is it documented to be a way of initializing the object? 

Would another alternative to be simply assign `PSA_MAC_OPERATION_INIT` to the struct?

Reply:
> Would another alternative to be simply assign `PSA_MAC_OPERATION_INIT` to the struct?

This line is an assignment, not a declarator, so the syntax used in the PSA specification would not work here. We would have to use a compound literal like below. But that is not one of the explicitly documented methods of initializing the structure.

```suggestion
	state->operation = (psa_mac_operation_t)PSA_MAC_OPERATION_INIT;
```

Language wise, it means the same thing and should work. But are we comfortable going outside what the crypto docs say is a 'must'? Maybe some compilers are non-compliant. I think the memset is safest.